src/report.rs

```diff
@@ -7,16 +7,26 @@ use error_stack::{AttachmentKind, Context, Frame, FrameKind, Report};
 use thiserror::Error;
 use valuable::Valuable;
 
-#[derive(Debug, Error)]
-#[error("{0}")]
-pub struct Error(String);
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("event sub failed")]
+    EventSub,
+    #[error("event processor failed")]
+    EventProcessor,
+    #[error("broadcast failed")]
+    Broadcast,
```

Comment:
```suggestion
    Broadcaster,
```
just to match the name of the actual thing